src/DataLayouts/cuda.jl

```diff
@@ -88,7 +100,11 @@ function Base.copyto!(
 ) where {S, Nij, A <: CUDA.CuArray}
     _, _, _, Nv, Nh = size(bc)
     if Nv > 0 && Nh > 0
-        CUDA.@cuda threads = (Nij, Nij) blocks = (Nh, Nv) knl_copyto!(dest, bc)
+        Nl = fld(CUDA.warpsize(), Nij * Nij)
```

Comment:
I think we can use up to `1024` threads. 
If we are hard-wired to use a `(Nij, Nij)` threadblock, we can just add an assertion `Nij * Nij <= 1024 (or max_threads)`. 
For more general code, `Nl` can be calculated inside the kernel! (`Nl = cld(Nij * Nij, blockDim.x()*blockDim.y()`)

Reply:
Good point. We probably need to play around with it and do some benchmarking to see what works.